soccer/gameplay/single_robot_sequence.py

```diff
@@ -0,0 +1,65 @@
+import behavior_sequence
+import single_robot_behavior
+import role_assignment
+
+#A sequence to be carried out by a single robot, attempting to add non-single robot behaviors will raise exceptons
+class SingleRobotSequence(behavior_sequence.BehaviorSequence):
```

Comment:
Python lets you inherit from multiple classes at once, so you can make SingleRobotSequence a subclass of both BehaviorSequence and SingleRobotCompositeBehavior.  If you do that, I think you can replace this whole class definition with:

```{.py}
class SingleRobotSequence(
    behavior_sequence.BehaviorSequence,
    single_robot_composite_behavior.SingleRobotCompositeBehavior):
    def __init__(self, behaviors=[], autorestart=lambda: True):
        single_robot_composite_behavior.SingleRobotCompositeBehavior.__init__(
            self,
            autorestart=autorestart)
        behavior_sequence.BehaviorSequence.__init__(self, behaviors=behaviors)

```

That way you get role_requirements(), assign_roles(), autorestart, etc for free. It looks like those methods are mostly the same as SingleRobotCompositeBehavior.